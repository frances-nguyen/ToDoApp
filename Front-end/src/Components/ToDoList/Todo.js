import React, { Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import TodoDataService from '../../API/TodoDataService'
import Authentication from '../AuthenticationService/Authentication'

class Todo extends Component {
    constructor(props) {
        super(props)
        this.state = {
            id: this.props.match.params.id,
            description: '',
            targetDate: moment(new Date()).format('YYYY-MM-DD')
        }
        this.onSubmit = this.onSubmit.bind(this)
        this.validate = this.validate.bind(this)
    }

    componentDidMount() {
        if (parseInt(this.state.id) === -1) {
            return
        }
        let username = Authentication.getUsername();
        TodoDataService.retrieveTodo(username, this.state.id)
            .then(response => {
                console.log(response.data.targetDate)
                this.setState({
                description: response.data.description,
                targetDate: moment(response.data.targetDate).format('YYYY-MM-DD')
            })})
    }

    validate(values) {
        let errors = {}
        if (!values.description) {
            errors.description = 'Enter a description.'
        } else if (values.description.length < 5) {
            errors.description = 'Enter atleast 5 characters in the description.'
        }

        if (!moment(values.targetDate, 'YYYY-MM-DD', true).isValid()) {
            errors.targetDate = 'Enter a valid target date.'
        }

        return errors
    }

    onSubmit(values) {
        let username = Authentication.getUsername();
        let todo = {
            id: this.state.id,
            description: values.description,
            targetDate: values.targetDate
        }

        if (this.state.id === -1) {
            TodoDataService.addTodo(username, todo)
                .then(() => this.props.history.push('/todolist'))
        } else {
            TodoDataService.updateTodo(username, this.state.id, todo)
                .then(() => this.props.history.push('/todolist'))
        }
    }

    render() {
        let { description, targetDate } = this.state

        return (
            <div>
                <div className="container">
                    <div className="row">
                        <h1 className="float-left">Todo</h1>
                    </div>
                    <div className="row">
                        <div className="col">
                            <Formik
                                initialValues={{ description, targetDate }}
                                onSubmit={this.onSubmit}
                                validate={this.validate}
                                validateOnChange={false}
                                validateOnBlur={false}
                                enableReinitialize={true}
                            >
                                {
                                    (props) => (
                                        <Form>
                                            <ErrorMessage
                                                name="description"
                                                component="div"
                                                className="alert alert-warning"
                                            />
                                            <ErrorMessage
                                                name="targetDate"
                                                component="div"
                                                className="alert alert-warning"
                                            />
                                            <fieldset className="form-group">
                                                <label className="float-left">Description</label>
                                                <Field
                                                    className="form-control"
                                                    type="text"
                                                    name="description">
                                                </Field>
                                            </fieldset>
                                            <fieldset className="form-group">
                                                <label className="float-left">Target Date</label>
                                                <Field
                                                    className="form-control"
                                                    type="date"
                                                    name="targetDate">
                                                </Field>
                                            </fieldset>
                                            <button className="btn btn-success float-left" type="submit">Save</button>
                                        </Form>
                                    )
                                }
                            </Formik>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

Todo.propTypes = {
    match: PropTypes.object
}

export default Todo