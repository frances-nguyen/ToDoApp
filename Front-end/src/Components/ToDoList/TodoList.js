import React, { Component } from 'react'
import TodoDataService from '../../API/TodoDataService'
import Authentication from '../AuthenticationService/Authentication'
import moment from 'moment'
import PropTypes from 'prop-types'

class TodoList extends Component {
  constructor(props) {
    super(props)
    this.state = {
      todos: [],
      message: null
    }
    this.addTodoClicked = this.addTodoClicked.bind(this)
    this.updateTodoClicked = this.updateTodoClicked.bind(this)
    this.deleteTodoClicked = this.deleteTodoClicked.bind(this)
    this.refreshTodos = this.refreshTodos.bind(this)
  }

  componentDidMount() {
    this.refreshTodos()
  }

  addTodoClicked() {
    this.props.history.push(`/todos/-1`)
  }

  updateTodoClicked(id) {
    this.props.history.push(`/todos/${id}`)
  }

  deleteTodoClicked(id) {
    let username = Authentication.getUsername()
    TodoDataService.deleteTodo(username, id)
      .then(response => {
        this.setState({ message: `Todo removed.` })
        this.refreshTodos()
        }
      )
  }

  refreshTodos() {
    let username = Authentication.getUsername()
    TodoDataService.retrieveAllTodos(username)
      .then(response =>
        this.setState({ todos: response.data }))
  }

  render() {
    return (
      <div>
        <div className="container">
          <div className="row">
            <h1 className="float-left">Todo</h1>
          </div>
          {this.state.message && <div className="alert alert-success">{this.state.message}</div>}
          <div className="row">
            <div className="table-responsive">
              <table className="table">
                <thead>
                  <tr>
                    <th scope="col">Description</th>
                    <th scope="col">Date</th>
                    <th scope="col">Complete</th>
                    <th scope="col">Update</th>
                    <th scope="col">Delete</th>
                  </tr>
                </thead>
                <tbody>
                  {this.state.todos.map(todo => (
                    <tr key={todo.id}>
                      <td>{todo.description}</td>
                      <td>{moment(todo.targetDate).format('MMMM Do, YYYY')}</td>
                      <td>{todo.completed.toString()}</td>
                      <td>
                        <button
                          className="btn btn-success"
                          onClick={() => this.updateTodoClicked(todo.id)}>Update
                    </button>
                      </td>
                      <td>
                        <button
                          className="btn btn-warning"
                          onClick={() => this.deleteTodoClicked(todo.id)}>Delete
                        </button>
                      </td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
          <div className="row">
            <button
              className="btn btn-success"
              onClick={this.addTodoClicked}>Add
            </button>
          </div>
        </div>
      </div>
    )
  }
}

TodoList.propTypes = {
  history: PropTypes.object
}

export default TodoList
