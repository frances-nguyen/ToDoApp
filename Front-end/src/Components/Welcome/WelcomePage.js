import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import HelloWorldService from '../../API/HelloWorldService'

class WelcomePage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      welcomeMessage: ''
    }
    this.retrieveWelcomeMessage = this.retrieveWelcomeMessage.bind(this)
    this.handleError = this.handleError.bind(this)
  }

  retrieveWelcomeMessage() {
    HelloWorldService.executeHelloWorldService()
      .then(response =>
        this.setState({ welcomeMessage: response.data }))
      .catch(error => this.handleError(error))
  }

  handleError(error) {
    let errorMessage = ''
    if (error.message) {
      errorMessage += error.message
    }
    if (error.response && error.response.data) {
      errorMessage += error.response.data.message
    }
    this.setState({ welcomeMessage: errorMessage })
  }

  render() {
    return (
      <div>
        <h1>Welcome!</h1>
        <div className="container">
          Welcome {this.props.match.params.name}. You can manage your to do list{' '}
          <Link to="/todolist">here</Link>.
        </div>
        <div className="container">
          Click here to get a customized welcome message.
          <button
            onClick={this.retrieveWelcomeMessage}
            className="btn btn-success">
            Get Welcome Message
          </button>
        </div>
        <div className="container">
          {this.state.welcomeMessage}
        </div>
      </div>
    )
  }
}

WelcomePage.propTypes = {
  match: PropTypes.object
}

export default WelcomePage
