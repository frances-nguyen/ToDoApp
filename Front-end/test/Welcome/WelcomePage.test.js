import React from 'react'
import renderer from 'react-test-renderer'
import WelcomePage from '../../src/Components/Welcome/WelcomePage'
import ToDoList from '../../src/Components/ToDoList/ToDoList'
import { MemoryRouter } from 'react-router-dom'

const match = {
  params: {
    name: 'username'
  }
}
describe('<WelcomePage /> rendering', () => {
  it('renders <WelcomePage />', () => {
    let tree = renderer
      .create(
        <MemoryRouter>
          <WelcomePage match={match} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
