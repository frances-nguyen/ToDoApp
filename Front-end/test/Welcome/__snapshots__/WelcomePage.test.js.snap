// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> rendering renders <WelcomePage /> 1`] = `
<div>
  Welcome 
  username
  . You can manage your to do list
   
  <a
    href="/todolist"
    onClick={[Function]}
  >
    here
  </a>
  .
</div>
`;
